MdeModulePkg/Include/Library/DeviceStateLib.h

```diff
@@ -32,27 +32,71 @@ MU_CHANGE: new file
 
 #define DEVICE_STATE_MAX  (1 << (31))
 
+#define DEVICE_STATE_PLATFORM_MODE_INDEX_START  20
+#define DEVICE_STATE_PLATFORM_MODE_INDEX_END    27

```

Comment:
Not a big deal, but personal preference is to directly associate the values with the reason they were chosen.

```suggestion
#define DEVICE_STATE_PLATFORM_MODE_INDEX_START  DEVICE_STATE_PLATFORM_MODE_0
#define DEVICE_STATE_PLATFORM_MODE_INDEX_END    DEVICE_STATE_PLATFORM_MODE_7
```

Reply:
This would change the values 
1 << 20 instead of 20. 
I did make a change to consume the macros for the start and end mode macros... not sure if that helps. Open to suggestions.